### DISCUSSION TITLE
KPZ equation with constant term

### DISCUSSION THREAD

**Discussion by ericdumonteil** [zwicker-group/py-pde#460] [Q&A] [answered]
Dear all,

I am trying to compare the KPZ equation to the Stochastic Heat equation (which are related by a Cole-Hopf transformation).
On this road, I need to numericaly solve the KPZ equation, to which a constant term is added.

<img width="274" alt="kpz+c" src="https://github.com/zwicker-group/py-pde/assets/66777598/c63bcf33-60b3-4134-a735-be83c00772cf">

I hence would like to overload the KPZInterface to add this constant, but I am failing to do so.

Does somebody have any hint how I could do that?

Thanks so much in advance,

Eric

**Comment by david-zwicker** [comment, marked as answer]
There are two simple solutions. You can either copy the [`KPZInterfacePDE` class of `py-pde`](https://github.com/zwicker-group/py-pde/blob/master/pde/pdes/kpz_interface.py) and simply change the respective terms. Alternatively, you can use the `PDE` class; see [the respective documentation](https://py-pde.readthedocs.io/en/latest/manual/basic_usage.html#specifying-the-pde). There is even [an example of the simple KPZ equation](https://py-pde.readthedocs.io/en/latest/examples_gallery/pde_custom_expression.html#sphx-glr-examples-gallery-pde-custom-expression-py) that you can augment.
